thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlDetectionAlertConfigTranslator.java

```diff
@@ -192,8 +195,17 @@ public DetectionAlertConfigDTO translate(Map<String,Object> yamlAlertConfig) {
     List<String> detectionNames = ConfigUtils.getList(yamlAlertConfig.get(PROP_DETECTION_NAMES));
 
     try {
-      detectionConfigIds.addAll(detectionNames.stream().map(detectionName ->  this.detectionConfigDAO.findByPredicate(
-          Predicate.EQ("name", detectionName)).get(0).getId()).collect(Collectors.toList()));
+      detectionConfigIds.addAll(detectionNames.stream().map(detectionName -> {
+        List<DetectionConfigDTO> detectionConfigDTOs = DAORegistry.getInstance().getDetectionConfigManager()
+            .findByPredicate(Predicate.EQ("name", detectionName));
+
+        if (!detectionConfigDTOs.isEmpty()) {
+          return detectionConfigDTOs.get(0).getId();
+        } else {
+          return DAORegistry.getInstance().getAnomalyFunctionDAO()
```

Comment:
Is it possible that we don't access the anomaly function DAO here? This translator will be kept after the migration, i.e. after the anomaly function DAO deprecated. I'd prefer not to refer it in the new code.